components/raftstore-v2/src/raft/peer.rs

```diff
@@ -849,4 +859,131 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             Err(e) => slog_panic!(self.logger, "failed to load term"; "index" => idx, "err" => ?e),
         }
     }
+
+    #[inline]
+    pub fn peer_status_context_mut(&mut self) -> &mut PeerStatusContext {
+        &mut self.peer_status_context
+    }
+
+    #[inline]
+    pub fn peer_status_context(&self) -> &PeerStatusContext {
+        &self.peer_status_context
+    }
+
+    /// Returns `true` if any peer recover from connectivity problem.
+    ///
+    /// A peer can become pending or down if it has not responded for a
+    /// long time. If it becomes normal again, PD need to be notified.
+    pub fn any_new_peer_catch_up(&mut self, peer_id: u64) -> bool {
```

Comment:
These methods and structs can be put in `raftstore-v2/src/operation/`.

Reply:
then how about move these code into `raftstore-v2/src/operation/pd`?

Reply:
Why this method is not moved?

Reply:
There are some references make it hard to move this function to `life.rs`. for example `self.raft_group`, `self.logger`. if we use closure here. the closure seems also complicated.